src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -587,6 +588,13 @@ def record_metadata_proportions(final_output_dir: Path) -> None:
         # or row noise type.
         melted_metadata["column"] = melted_metadata["variable"].str.split(".").str[0]
         melted_metadata["noise_type"] = melted_metadata["variable"].str.split(".").str[1]
+        # Swap in correct number of rows for guardian duplication to calculate proportions
+        if dataset in [
+            metadata.DatasetNames.CENSUS,
+            metadata.DatasetNames.ACS,
+            metadata.DatasetNames.CPS,
+        ]:
+            melted_metadata = update_guardian_duplication_row_counts(melted_metadata)
```

Comment:
I don't understand the motivation for this change. What exactly is this fixing?

Reply:
Basically guardian duplication configuration values are conditional probabilities meaning its the probability you are in a one of the two groups and live at a different address from one of your guardians given you are in one of the groups. The error Im fixing is we get "number of rows" which is the length of our dataframe but for guardian duplication we need to get the number of rows for each group. So in one of the helper functions we capture that value and then in this function we swap that out with number of rows.